packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx

```diff
@@ -10,48 +10,85 @@ export const useToggleButtonClasses = makeVariantClasses<ToggleButtonState>({
+        background: 'var(--button-checked-pressed-background, var(--button-checked-hovered-background))',
+        color: 'var(--button-checked-pressed-contentColor, var(--button-checked-hovered-contentColor))',
+        '.ms-Button-icon': {
+          color: 'var(--button-checked-pressed-iconColor, var(--button-checked-hovered-iconColor))',
+        },
       },
     },
   },
 
-  variants: {
-    primary: {
-      checked: {
-        background: 'var(--color-brand-checked-background)',
-        contentColor: 'var(--color-brand-checked-contentColor)',
-      },
-      checkedHovered: {
-        background: 'var(--color-brand-checkedHovered-background)',
-        contentColor: 'var(--color-brand-checkedHovered-contentColor)',
-      },
-      checkedPressed: {
-        background: 'var(--color-brand-checkedPressed-background)',
-        contentColor: 'var(--color-brand-checkedPressed-contentColor)',
+  variants: (theme: Theme) => {
+    const { palette, semanticColors } = theme;
+
+    return {
+      root: {
+        checked: {
+          background: semanticColors?.buttonBackgroundPressed,
+          contentColor: semanticColors?.buttonTextChecked,
+
+          hovered: {
+            background: semanticColors?.buttonBackgroundPressed,
+            contentColor: semanticColors?.buttonTextCheckedHovered,
+          },
+        },
       },
-    },
 
-    ghost: {
-      checked: {
-        background: 'var(--ghost-checked-background)',
-        contentColor: 'var(--ghost-checked-contentColor)',
+      primary: {
+        checked: {
+          background: 'var(--color-brand-checked-background)',
+          contentColor: 'var(--color-brand-checked-contentColor)',
+          hovered: {
+            background: 'var(--color-brand-checked-hovered-background)',
+            contentColor: 'var(--color-brand-checked-hovered-contentColor)',
+          },
+          pressed: {
+            background: 'var(--color-brand-checked-pressed-background)',
+            contentColor: 'var(--color-brand-checked-pressed-contentColor)',
+          },
+        },
       },
-      checkedHovered: {
-        background: 'var(--ghost-checkedHovered-background)',
-        contentColor: 'var(--ghost-checkedHovered-contentColor)',
+
+      ghost: {
+        checked: {
+          background: palette?.neutralLight,
+          contentColor: palette?.black,
+          iconColor: palette?.themeDarker,
+
+          hovered: {
+            background: palette?.neutralQuaternaryAlt,
+            contentColor: palette?.themePrimary,
+            iconColor: palette?.themePrimary,
+          },
+        },
       },
-      checkedPressed: {
-        background: 'var(--ghost-checkedPressed-background)',
-        contentColor: 'var(--ghost-checkedPressed-contentColor)',
+
+      transparent: {
+        checked: {
+          background: 'transparent',
+          contentColor: palette?.themePrimary,
+          iconColor: palette?.themePrimary,
+          hovered: {
```

Comment:
this should be collapsed as `checkedHovered`.

Reply:
See `ColorTokens` in Theme. This should really be complaining in the typings. Can you see if you can make this typesafe?

Reply:
Spoke with @dzearing offline, I will address collapsing things in this PR but I'll check the typings in a different PR.